pkg/proxmox/goproxmox/api_client.go

```diff
@@ -141,6 +142,51 @@ func (c *APIClient) FindVMResource(ctx context.Context, vmID uint64) (*proxmox.C
 	return nil, fmt.Errorf("unable to find VM with ID %d on any of the nodes", vmID)
 }
 
+// FindVMTemplateByTags tries to find a VMID by its tags across the whole cluster.
+func (c *APIClient) FindVMTemplateByTags(ctx context.Context, templateTags []string) (string, int32, error) {
```

Comment:
Hm, `templateTags` could be `nil` here:`GetTemplateSelectorTags` can return `nil` but its result is passed to `FindVMTemplateByTags` https://github.com/ionos-cloud/cluster-api-provider-proxmox/pull/343/files#diff-1a30de7fe02a7f3127edda65e72bcd4b7a77ac737126aed0fe007d3cfba9d224R389-R390

Can you add a test that calls `FindVMTemplateByTags(..., nil)`, please?